KG-509: Add error handling for missing tools in GenericAgentEnvironment

denis-domanskii opened the pull request:
<!--
Thank you for opening a pull request!

Please add a brief description of the proposed change here.
Also, please tick the appropriate points in the checklist below.
-->

## Motivation and Context
https://youtrack.jetbrains.com/issue/KG-509/An-exception-when-a-tool-doesnt-exist
## Breaking Changes
<!-- Will users need to update their code or configurations? -->

---

#### Type of the changes
- [ ] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update
- [ ] Tests improvement
- [ ] Refactoring

#### Checklist
- [x] The pull request has a description of the proposed change
- [x] I read the [Contributing Guidelines](https://github.com/JetBrains/koog/blob/main/CONTRIBUTING.md) before opening the pull request
- [x] The pull request uses **`develop`** as the base branch
- [x] Tests for the changes have been added
- [x] All new and existing tests passed

##### Additional steps for pull requests adding a new feature
- [ ] An issue describing the proposed change exists
- [ ] The pull request includes a link to the issue
- [ ] The change was discussed and approved in the issue
- [ ] Docs have been added / updated

denis-domanskii committed:
KG-509: Add error handling for missing tools in GenericAgentEnvironment

EugeneTheDev commented on agents/agents-core/src/commonMain/kotlin/ai/koog/agents/core/environment/GenericAgentEnvironment.kt:
```diff
@@ -90,7 +90,18 @@ internal class GenericAgentEnvironment(
-        val tool = toolRegistry.getTool(content.toolName)
+        val tool = try {
+            toolRegistry.getTool(content.toolName)
+        } catch (e: Exception) {
```
Catching generic exception smells a bit. Let's add getToolOrNull method to ToolRegistry instead?

denis-domanskii committed:
KG-509: Add graceful handling for non-existent tools in agent environment and tests

### Key Updates
- Introduced `getToolOrNull` method for retrieving tools without throwing exceptions.
- Updated `GenericAgentEnvironment` to handle missing tools gracefully.
- Expanded tests to verify proper error handling and LLM-requested tool tracking.
- Enhanced testing logic to ensure agents complete execution without exceptions.
